File: 728-self-dividing-numbers/self-dividing-numbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> list=new ArrayList<>();
        for(int i=left;i<=right;i++){
            if(i<=9) {
                list.add(i);
                continue;
            }
            if(divide(i)){
                list.add(i);
            }
        }
        return list;
    }
    public static boolean divide(int num){
        int or=num;
        while(num!=0){
            int rev=num%10;
            if(rev==0 || or%rev !=0)  return false;
            num/=10;
        }
        return true;
    }
}

File: 728-self-dividing-numbers/self-dividing-numbers.py
class Solution(object):
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        def isdivide(num):
            org=num
            while num!=0:
                rem=num%10
                if rem==0 or org % rem != 0:
                    return False
                num//=10
            return True
                                                                                                                          
        lst=[]
        for i in range(left,right+1):
            if i<=9:
                lst.append(i)
                continue
            if isdivide(i):
                lst.append(i)
        
        return lst


        
        